### PR TITLE
feat: Support ClickHouse Grammar

### PR DESCRIPTION
**Pull request by hanyuancheung**
- Support ClickHouse basic grammar.

### CODE REVIEW

**Comment by hanyuancheung** [comment]
hi @huandu 
I've found that sqlbuilder still not support clickhouse grammar, so I contribute this part to use this tool to generate sql statement. Could you please to review it, thanks.

File: interpolate.go [hunk 1]
```diff
@@ -521,6 +528,10 @@ func encodeValue(buf []byte, arg interface{}, flavor Flavor) ([]byte, error) {
 			case SQLServer:
 				buf = append(buf, "0x"...)
 				buf = appendHex(buf, data)
+
+			case ClickHouse:
+				buf = append(buf, "_binary"...)
```
Comment (huandu) [inline interpolate.go:533]: I don't find `_binary` method in ClickHouse doc. Instead, I find [`unhex`](https://clickhouse.com/docs/en/sql-reference/functions/encoding-functions#unhex) seems the most appropriate method to convert arbitrary data to blob. You can try to write logic against `unhex`, or you can provide me a pointer to `_binary` if this method works well in ClickHouse.

File: flavor.go [hunk 2]
```diff
@@ -146,6 +151,9 @@ func (f Flavor) PrepareInsertIgnore(table string, ib *InsertBuilder) {
 	case SQLite:
 		// see https://www.sqlite.org/lang_insert.html
 		ib.verb = "INSERT OR IGNORE"
+	case ClickHouse:
+		// see https://clickhouse.tech/docs/en/sql-reference/statements/insert-into/
+		ib.verb = "INSERT"
```
Comment (huandu) [inline flavor.go:156]: The verb should be `INSERT INTO`. And it also should be `INSERT INTO` in PostgreSQL flavor -- it should be an aged bug.

**Review by huandu — CHANGES_REQUESTED** [review]

**Comment by huandu** [comment]
Thanks for your contribution! Please review my feedback and make changes. Feel free to discuss with me if you have any question.

**Comment by hanyuancheung** [comment]
> Thanks for your contribution! Please review my feedback and make changes. Feel free to discuss with me if you have any question.

Thanks for ur reply. I've fixed these comments.

File: interpolate.go [hunk 3]
```diff
@@ -521,6 +528,10 @@ func encodeValue(buf []byte, arg interface{}, flavor Flavor) ([]byte, error) {
 			case SQLServer:
 				buf = append(buf, "0x"...)
 				buf = appendHex(buf, data)
+
+			case ClickHouse:
+				buf = append(buf, "unhex"...)
+				buf = quoteStringValue(buf, *(*string)(unsafe.Pointer(&data)), flavor)
```
Comment (huandu) [inline interpolate.go:534]: Call `appendHex` instead `quoteStringValue`.

File: interpolate_test.go [hunk 4]
```diff
@@ -163,6 +163,52 @@ func TestFlavorInterpolate(t *testing.T) {
...
+		},
+		{
+			ClickHouse,
+			"SELECT * FROM `a?` WHERE name = \"?\" AND state IN (?, '?', ?, ?, ?, ?, ?)", []interface{}{"\r\n\b\t\x1a\x00\\\"'", uint(42), uint8(8), uint16(16), uint32(32), uint64(64), "useless"},
+			"SELECT * FROM `a?` WHERE name = \"?\" AND state IN ('\\r\\n\\b\\t\\Z\\0\\\\\\\"\\'', '?', 42, 8, 16, 32, 64)", nil,
+		},
+		{
+			ClickHouse,
+			"SELECT ?, ?, ?, ?, ?, ?, ?, ?, ?", []interface{}{true, false, float32(1.234567), 9.87654321, []byte(nil), []byte("I'm bytes"), dt, time.Time{}, nil},
+			"SELECT TRUE, FALSE, 1.234567, 9.87654321, NULL, _binary'I\\'m bytes', '2019-04-24 12:23:34.123457', '0000-00-00', NULL", nil,
```
Comment (huandu) [inline interpolate_test.go:180]: Don't forget to update case to cover `[]byte` type. And try the expected SQL in your ClickHouse environment to make sure the SQL really works well.

**Review by huandu — CHANGES_REQUESTED** [review]
There are still some blocking issues. Please fix them. Thanks.

**Comment by hanyuancheung** [comment]
> There are still some blocking issues. Please fix them. Thanks.

Sure, I've fixed them and tested in my local clickhouse. It works well. Thanks for the advice.

**Review by huandu — APPROVED** [review]
LGTM.

**Pull request merged**